azurerm/internal/services/cdn/cdn_endpoint_resource.go

```diff
@@ -447,6 +448,10 @@ func resourceArmCdnEndpointRead(d *schema.ResourceData, meta interface{}) error
 		d.Set("probe_path", props.ProbePath)
 		d.Set("optimization_type", string(props.OptimizationType))
 
+		if is_compression_enabled := props.IsCompressionEnabled; is_compression_enabled != nil {
+			d.Set("is_compression_enabled", *is_compression_enabled)
+		}
```

Comment:
The reason why the test is failing is because the value is always returned by the API, so we need to check the config file again in the `Read` function to determine if we should write the value to the state file or not, something like this:

```suggestion
		if _, ok := d.GetOk("is_compression_enabled"); ok {
			if compressionEnabled := props.IsCompressionEnabled; compressionEnabled != nil {
				d.Set("is_compression_enabled", *compressionEnabled)
			}
		}
```